guppylang/tys/parsing.py

```diff
@@ -138,10 +190,29 @@ def type_with_flags_from_ast(
     globals: Globals,
     param_var_mapping: dict[str, Parameter] | None = None,
 ) -> tuple[Type, InputFlags]:
-    """Turns an AST expression into a Guppy type possibly annotated with @flags."""
-    # Parse an argument and check that it's valid for a `TypeParam`
-    arg, flags = arg_from_ast(node, globals, param_var_mapping)
-    return _type_param.check_arg(arg, node).ty, flags
+    """Turns an AST expression into a Guppy type with some optional @flags."""
+    # Check for `type @flag` annotations
+    if isinstance(node, ast.BinOp) and isinstance(node.op, ast.MatMult):
+        ty, flags = type_with_flags_from_ast(node.left, globals, param_var_mapping)
```

Comment:
Note this means `x @inout @inout` is also allowed and has the same effect as only once

Reply:
Indeed, I think that should be fine though